types/plaid-link/index.d.ts

```diff
@@ -0,0 +1,117 @@
+// Type definitions for non-npm package plaid-link-browser 2.0
+// Project: https://github.com/plaid/link (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
+// Definitions by: Aaron Holderman <https://github.com/afholderman>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped/
+// TypeScript Version: 2.4
+
+/*~ If this module is a UMD module that exposes a global variable 'myLib' when
+ *~ loaded outside a module loader environment, declare that global here.
+ *~ Otherwise, delete this declaration.
+ */
+
+declare global {
+    interface Window {
+        Plaid: {
+            create(params: Plaid.CreateConfig): Plaid.LinkHandler;
+            open(): void;
+            exit(force?: boolean): void;
+        };
+        linkHandler: Plaid.LinkHandler;
+    }
+}
+
+export namespace Plaid {
```

Comment:
Since you're making a global value declaration, why not just make `Plaid` a global namespace that exports 3 functions?

Reply:
I'm sorry I don't understand this request, do you have an example of what you're looking for?

Reply:
Sure - would it make sense to get rid of the `declare global` block and write the namespace as a global with 3 functions?

```ts
namespace Plaid {
    export function create(params: Plaid.CreateConfig): Plaid.LinkHandler;
    export function open(): void;
    export function exit(force?: boolean): void;

    // ...

    type EventName =
        | "ERROR"
        // ...
}
```

Reply:
I tested this and it does not work as expected. Plaid-link loads via a script and is made available on the window object. Without out it in the declare global Window TS throws: Property 'Plaid' does not exist on type 'Window'.

Reply:
Here is a sandbox with an example implementation: https://codesandbox.io/s/8k2qll1900?fontsize=14&module=%2Fsrc%2FPlaidComponent%2FPlaidComponent.tsx

Reply:
Hey @DanielRosenwasser if you get a chance mind looking at the above?